lib/activerecord-import/import.rb

```diff
@@ -424,21 +431,24 @@ def import_from_table( options ) # :nodoc:
     def import_with_validations( column_names, array_of_attributes, options = {} )
       failed_instances = []
 
-      # create instances for each of our column/value sets
-      arr = validations_array_for_column_names_and_attributes( column_names, array_of_attributes )
+      if block_given?
+        yield failed_instances
+      else
+        # create instances for each of our column/value sets
+        arr = validations_array_for_column_names_and_attributes( column_names, array_of_attributes )
 
-      # keep track of the instance and the position it is currently at. if this fails
-      # validation we'll use the index to remove it from the array_of_attributes
-      arr.each_with_index do |hsh, i|
-        instance = new do |model|
+        # keep track of the instance and the position it is currently at. if this fails
+        # validation we'll use the index to remove it from the array_of_attributes
+        model = new
+        arr.each_with_index do |hsh, i|
           hsh.each_pair { |k, v| model[k] = v }
```

Comment:
@zdennis instead of instantiating a new instance for every record, here I changed it to instantiate one instance and just update the attributes on that model.